Exiled.CustomItems/Events/PlayerHandler.cs

```diff
@@ -0,0 +1,55 @@
+// -----------------------------------------------------------------------
+// <copyright file="PlayerHandler.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.CustomItems
+{
+    using Exiled.API.Extensions;
+    using Exiled.API.Features;
+    using Exiled.CustomItems.API.Features;
+    using Exiled.Events.EventArgs;
+
+    /// <summary>
+    /// Handles Player events for the CustomItem API.
+    /// </summary>
+    internal sealed class PlayerHandler
+    {
+        public void OnChangingRole(ChangingRoleEventArgs ev)
+        {
+            switch (ev.NewRole == RoleType.Spectator)
+            {
+                case true:
+                {
+                    foreach (Player player in Player.List)
+                    {
+                        if (player == ev.Player)
+                            continue;
+
+                        if (CustomItem.TryGet(player, out CustomItem item))
+                        {
+                            ev.Player.SendFakeSyncVar(player.ReferenceHub.networkIdentity, typeof(NicknameSync), nameof(NicknameSync.Network_myNickSync), $"{ev.Player.Nickname} (CustomItem: {item.Name})");
+                        }
+                    }
+
+                    break;
+                }
+
+                case false:
+                {
+                    foreach (Player player in Player.List)
+                    {
+                        if (player == ev.Player)
+                            continue;
+
+                        ev.Player.SendFakeSyncVar(player.ReferenceHub.networkIdentity, typeof(NicknameSync), nameof(NicknameSync.Network_myNickSync), player.Nickname);
```

Comment:
Couldn't `ResyncSyncVar` be used like above?
Also, this check should only be done on "hacky" custom items.

Reply:
Yeah, I was thinking about using it instead of the foreach.

Reply:
But this is in OnChangingRole, I only want the person who is becoming non-spectator to have the name re-synced. ResyncSyncVar would reset the name for all players, so a foreach is needed here.

Reply:
Like in the other comment, I thought it was inside CustomItem class.